pkg/utils/reconcilename/reconcilename.go

```diff
@@ -0,0 +1,33 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilename
```

Comment:
Can we merge all of these into a single injection package? I really like the knative approach I linked. I think it'll cause this to read really nicely.

```
injection.WithFoo(ctx, foo)
injection.GetFoo(ctx)
```